ytypes/int_type_test.go

```diff
@@ -307,6 +341,75 @@ func TestValidateInt(t *testing.T) {
 	}
 }
 
+func TestValidateUint(t *testing.T) {
+	tests := []struct {
+		desc      string
+		ranges    yang.YangRange
+		inValues  []int64
+		outValues []int64
+	}{
+		{
+			desc: "single val range 0",
+			ranges: yang.YangRange{
+				yang.YRange{Min: yang.FromInt(0), Max: yang.FromInt(0)},
+			},
+			inValues:  []int64{0},
+			outValues: []int64{1},
+		},
+		{
+			desc: "single val range +ve",
+			ranges: yang.YangRange{
+				yang.YRange{Min: yang.FromInt(10), Max: yang.FromInt(10)},
+			},
+			inValues:  []int64{10},
+			outValues: []int64{0, 9, 11},
+		},
+		{
+			desc: "ranges [0,+], [+,+]",
+			ranges: yang.YangRange{
+				yang.YRange{Min: yang.FromInt(0), Max: yang.FromInt(3)},
+				yang.YRange{Min: yang.FromInt(5), Max: yang.FromInt(10)},
+			},
+			inValues:  []int64{1, 2, 3, 5, 7, 10},
+			outValues: []int64{4, 11},
+		},
+		{
+			desc: "ranges [0,+], [+,+inf]",
+			ranges: yang.YangRange{
+				yang.YRange{Min: yang.FromInt(0), Max: yang.FromInt(3)},
+				yang.YRange{Min: yang.FromInt(5), Max: yang.FromInt(10)},
+			},
+			inValues:  []int64{0, 1, 2, 3, 5, 7, 10},
+			outValues: []int64{4, 11},
+		},
+	}
+
+	//yangIntTypes := []yang.TypeKind{yang.Yuint8, yang.Yuint16, yang.Yuint32, yang.Yuint64}
```

Comment:
Uncomment this line and remove the one below? Looks like this one gives wider sets of tests.

Reply:
Done.